src/test/java/net/minecraftforge/debug/fluid/CoreFluidTest.java

```diff
@@ -0,0 +1,355 @@
+            {
+                for (Direction direction : Direction.values())
+                {
+                    if (direction != Direction.DOWN)
+                    {
+                        if (level.getFluidState(pos.relative(direction)).getType() == Fluids.LAVA)
+                        {
+                            level.setBlock(pos, level.getBlockState(pos.below()), Constants.BlockFlags.DEFAULT);
+                            return false;
+                        }
+                    }
+                }
+                return true;
+            }
+        }
+
+        public static class Flowing extends ForgeFlowingFluid.Flowing
+        {
+            public Flowing(Properties properties)
+            {
+                super(properties);
+            }
+
+            @Override
+            public boolean canSwim(FluidState state)
+            {
+                return true;
+            }
+
+            @Override
+            public boolean canDrown(FluidState state, LivingEntity entity)
+            {
+                return entity instanceof Player;
+            }
+
+            @Override
+            public boolean canExtinguish(FluidState state, Entity entity)
+            {
+                return entity instanceof Player;
+            }
+
+            @Override
+            public boolean canHydrate(FluidState fluidState, BlockState blockState)
+            {
+                return blockState.getBlock() instanceof ConcretePowderBlock || blockState.getBlock() instanceof FarmBlock;
+            }
+
+            @Override
+            public boolean canBoat(FluidState state, Boat boat)
+            {
+                return false;
+            }
+
+            @Override
+            public void sink(FluidState state, LivingEntity entity) {
+                entity.setDeltaMovement(entity.getDeltaMovement().add(0.0D, (double)-0.1F * entity.getAttribute(net.minecraftforge.common.ForgeMod.SWIM_SPEED.get()).getValue(), 0.0D));
+            }
+
+            // If the custom fluid is trying to place a flowing block, and it touches a Lava block.
+            // Then it will generate a copy of the BlockState below said position.
+            @Override
+            public boolean handleFluidInteraction(FluidState state, Level level, BlockPos pos)
+            {
+                for (Direction direction : Direction.values())
+                {
+                    if (direction != Direction.DOWN)
+                    {
+                        if (level.getFluidState(pos.relative(direction)).getType() == Fluids.LAVA)
+                        {
+                            level.setBlock(pos, level.getBlockState(pos.below()), Constants.BlockFlags.DEFAULT);
+                            return false;
+                        }
+                    }
+                }
+                return true;
+            }
+        }
+    }
+
+    private static class FluidloggableBlock extends Block implements SimpleWaterloggedBlock
```

Comment:
I would say given this is an example for people this really should implement `BucketPickup` and `LiquidBlockContainer` instead of `SimpleWaterloggedBlock` as it doesn't actually support water and all the methods except `getPickupSound` are being overridden anyways. That way if anyone copies this style of doing it into their mod they don't run into any issues if another mod is doing an instanceof check on SimpleWaterloggedBlock (I vaguely recall running into at least one mod doing that at some point)